invenio_queues/__init__.py

```diff
@@ -6,7 +6,149 @@
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
-"""Invenio module for collecting statistics."""
+
+r"""Invenio module for managing queues.
+
+This guide will show you how to get started with Invenio-Queues. It assumes
+that you already have knowledge of Flask applications and Invenio modules.
+
+It will then explain key topics and concepts of this module.
+
+Getting started
+---------------
+
+You will learn how to register queues and to interact with it. To begin with,
+we assume that you already have setup your virtual environment and install
+the module:
+
+You now need an application to work with, that can be created with the
+following commands in a Python shell:
+
+.. code-block:: python
+
+   from flask import Flask
+   app = Flask('myapp')
+
+You can add the queue in the configuration:
+
+.. code-block:: python
+
+   app.config['QUEUES_DEFINITIONS'] = [
+       {
+           'name': 'notifications',
+           'exchange': Exchange(
+               'example',
+               type='direct',
+               delivery_mode='transient',  # in-memory queue
+           ),
+       },
+   ]
+
+You can then initialize the module:
+
+.. code-block:: python
+
+   from invenio_queues.ext import InvenioQueues
+   ext_queues = InvenioQueues(app)
+
+In our example, we are using RabbitMQ as a broker, which can be configured
+as follow::
+
+    app.config['QUEUES_BROKER_URL'] = 'amqp://localhost:5672'
+
+Register queues
+^^^^^^^^^^^^^^^
+
+To register queues, you need to start by creating an exchange
+(https://docs.celeryproject.org/projects/kombu/en/stable/reference/kombu.html#exchange)
+for the queues:
+
+.. code-block:: python
+
+   from kombu import Exchange
+
+   default_exchange = Exchange(
+       'example',
+       type='direct',
+       delivery_mode='transient',  # in-memory queue
+   )
```

Comment:
You can move this above, in the section that you set the `QUEUES_DEFINITIONS` config